packages/airnode-protocol/contracts/rrp/authorizers/admin/AirnodeTokenPayment.sol

```diff
@@ -0,0 +1,286 @@
+    /// @param _requesterAddress Requester address
+    /// @param _whitelistDuration Duration in seconds for which the requester
+    /// will be whitelisted
+    function makePayment(
+        uint256 _chainId,
+        address _airnode,
+        bytes32 _endpointId,
+        address _requesterAddress,
+        uint64 _whitelistDuration
+    ) external override {
+        require(_chainId != 0, "Invalid chainId");
+        require(_airnode != address(0), "Invalid Airnode address");
+        require(_requesterAddress != address(0), "Invalid requester address");
+        require(
+            _whitelistDuration != 0 &&
+                _whitelistDuration <= DEFAULT_MAXIMUM_WHITELIST_DURATION &&
+                _whitelistDuration >= getMinimumWhitelistDuration(_airnode),
+            "Invalid whitelist duration"
+        );
+
+        address requesterAuthorizerWithManager = IAirnodeRequesterAuthorizerRegistry(
+                airnodeRequesterAuthorizerRegistry
+            ).chainIdToRequesterAuthorizerWithManager(_chainId);
+        require(
+            requesterAuthorizerWithManager != address(0),
+            "No RequesterAuthorizersWithManager set for chainId"
+        );
+
+        (
+            uint64 expirationTimestamp,
+            uint192 indefiniteWhitelistCount
+        ) = IRequesterAuthorizerWithManager(requesterAuthorizerWithManager)
+                .airnodeToEndpointIdToRequesterToWhitelistStatus(
+                    _airnode,
+                    _endpointId,
+                    _requesterAddress
+                );
+        require(
+            indefiniteWhitelistCount == 0,
+            "Already whitelisted indefinitely"
+        );
+        require(
+            uint64(block.timestamp) + _whitelistDuration > expirationTimestamp,
+            "Already whitelisted"
+        );
+
+        uint256 amount = getPaymentAmount(
+            _chainId,
+            _airnode,
+            _endpointId,
+            expirationTimestamp > uint64(block.timestamp)
+                ? (uint64(block.timestamp) + _whitelistDuration) -
+                    expirationTimestamp
+                : _whitelistDuration
+        );
+
+        IRequesterAuthorizerWithManager(requesterAuthorizerWithManager)
+            .extendWhitelistExpiration(
+                _airnode,
+                _endpointId,
+                _requesterAddress,
+                uint64(block.timestamp) + _whitelistDuration
+            );
+
+        assert(
+            IERC20Metadata(paymentTokenAddress).transferFrom(
+                msg.sender,
+                getAirnodeToPaymentDestination(_airnode),
+                amount
+            )
+        );
+
+        emit MadePayment(
+            _chainId,
+            _airnode,
+            _endpointId,
+            _requesterAddress,
+            msg.sender,
+            getAirnodeToPaymentDestination(_airnode),
+            amount,
+            IERC20Metadata(paymentTokenAddress).symbol(),
```

Comment:
I think address is smaller than symbol so might be better to use that?

Reply:
The think here is that in the event I think it might be more useful to have the symbol instead. Don't you agree? Should I add both maybe?